src/gt4py/next/ffront/func_to_past.py

```diff
@@ -162,6 +162,19 @@ def visit_BinOp(self, node: ast.BinOp, **kwargs: Any) -> past.BinOp:
     def visit_Name(self, node: ast.Name) -> past.Name:
         return past.Name(id=node.id, location=self.get_location(node))
 
+    def visit_Attribute(self, node: ast.Attribute) -> past.AttributeExpr:
+        attr_type = (
+            type_translation.from_value(self.closure_vars[node.attr])
+            if node.attr in self.closure_vars
+            else None
+        )
+        return past.AttributeExpr(
+            attr=node.attr,
+            value=self.visit(node.value),
+            type=attr_type,
```

Comment:
Isn't that already
```suggestion
            type=self.closure_vars[node.attr].type,
```

Reply:
In this specific case `self.closure_vars[node.attr]` is:
`Dimension(value='IDim', kind=<DimensionKind.HORIZONTAL: 'horizontal'>)`
Hence it does not have a type attribute. Also `attr_type` is needed because of the lines above, i.e.
```
        attr_type = (
            type_translation.from_value(self.closure_vars[node.attr])
            if node.attr in self.closure_vars
            else None
        )
```

Reply:
- Actually, I realize now, the type should be only deduced in type_deduction, so here it should be left empty.
- But, the type here is actually wrong, because `self.closure_vars[node.attr]` is looking for a top-level name `node.attr`, that's why the tests break if you remove the top-level import, see my comment in the test. Correct would be to do something like `getattr(self.closure_vars[node.value.id], node.attr)`.